generate.py

```diff
@@ -124,8 +124,9 @@ def main(
     t0 = time.time()
     with EmptyInitOnDevice(device=fabric.device, dtype=dtype, quantization_mode=quantize):
         model = Parrot(config)
-    with lazy_load(checkpoint_path) as checkpoint:
-        model.load_state_dict(checkpoint)
+
+        with lazy_load(checkpoint_path) as checkpoint:
```

Comment:
The question is why the KeyError appears. For instance, Lit-LLaMA doesn't need `strict=False`: https://github.com/Lightning-AI/lit-llama/blob/main/generate.py#L134

Reply:
> For "llm.int8", when we load checkpoint outside EmptyInit with strict=False, the model is not loaded properly. It outputs gibberish.

I tried indenting and unindenting and the output looks the same to me.

Reply:
With load checkpoint inside EmptyInit with strict=False it looks work properly
Here is my output:

`Loading model 'checkpoints/stabilityai/stablelm-base-alpha-3b/lit_model.pth' with {'block_size': 4096, 'vocab_size': 50254, 'padding_multiple': 512, 'padded_vocab_size': 50688, 'n_layer': 16, 'n_head': 32, 'n_embd': 4096, 'rotary_percentage': 0.25, 'parallel_residual': True}
Time to load model: 6.52 seconds.
Global seed set to 1234
Hello, my name is Cody, and I write to you...
I need a police officer to stop shining light on troubled individuals in exchange for a lighter sentence for a crime.
I need a police officer to stop shining light on troubled individuals in exchange for a lighter sentence
Time for inference 1: 2.56 sec total, 19.54 tokens/sec
Memory used: 5.20 GB
`

Reply:
@ArturK-85 Do you see a difference with loading inside EmptyInit vs outside? (both with `strict=False`)

Subjectively, the answer looks fine considering that these checkpoints are not very powerful.

Reply:
I don't see any difference when loading inside/outside context manager both with strict=False

Reply:
@carmocca 
Somehow I can't replicate difference with loading inside EmptyInit vs outside. Maybe I made some mistakes.



However, I found the root of error for `llm.int8`.

https://github.com/Lightning-AI/lit-parrot/blob/f6e17d40a4a2aa77ad4d8659411b59a0ee61c1f4/lit_parrot/quantization.py#L62-L64

When we let nn.Module load the bias. The method expects both weight and bias, but the weight has been popped off the `local_state_dict`.


Why Lit-LLaMA doesn't has this error? Because the model doesn't have bias parameters according to the `convert_hf_checkpoint` script.

Reply:
Nice find @bkiat1123